heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -229,6 +231,38 @@
   public static final String TOPOLOGY_UPDATE_REACTIVATE_WAIT_SECS =
       "topology.update.reactivate.wait.secs";
 
+  /**
+   * Packing algorithm used to calculate packing plan
+   */
+  public static final String TOPOLOGY_PACKING_ALGORITHM =
```

Comment:
I really doubt if we need the packing algorithm to be configurable by users. By the original design, it's a system wide property which should only be set by the heron administrator.

Also, how are the `ram`, `cpu` and `disk` resource configs used? It seems these config doesn't tell the difference for different components. It applies all components' instances for a topology. Correct me if my understanding is wrong here.